geom/type_geometry_collection.go

```diff
@@ -212,6 +212,24 @@ func (c GeometryCollection) MarshalJSON() ([]byte, error) {
 	return buf, nil
 }
 
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface by decoding
```

Comment:
These implementations are pretty boilerplatey... I don't see an easy way around it though.

Reply:
I tried using reflect to remove the boilerplate (the approach works -- unit tests still pass). However, it's pretty hard to read, and overall I _think_ it's probably not an improvement even though it does successfully eliminate 90% of the boilerplate. I've reverted the change, but would like to get anyone's opinion on it. 7d07b3852658d81e3804113e761c1ccf1faddaf3

Reply:
I've got a new approach to reduce boilerplate here: a855eacfeec8029626b9988282f9c251b5db46dd It follows the same strategy taken for WKB scanning boilerplate (and even shares some of the type dispatch code). I think that's a bit cleaner, and doesn't rely on reflect.